predicators/envs/cover.py

```diff
@@ -364,6 +364,130 @@ def _any_intersection(self,
         return False
 
 
+class CoverEnvHandEmpty(CoverEnv):
+    """Toy cover domain where the robot has a feature indicating whether its
+    hand is empty or not.
+
+    This allows us to learn all the predicates (Cover, Holding,
+    HandEmpty) through interactive learning.
+    """
+
+    def __init__(self, use_gui: bool = True) -> None:
+        super().__init__(use_gui)
+
+        # Add attribute.
+        self._robot_type = Type("robot",
+                                ["hand", "pose_x", "pose_z", "hand_empty"])
+        # Override HandEmpty predicate.
+        self._HandEmpty = Predicate("HandEmpty", [self._robot_type],
+                                    self._HandEmpty_holds)
+        # Create new robot because of new robot type
+        self._robot = Object("robby", self._robot_type)
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "cover_handempty"
+
+    def simulate(self, state: State, action: Action) -> State:
```

Comment:
please refactor this and `_create_initial_state` to remove as much redundant code as possible 😇 the changes w.r.t. to the parent class should be <10 lines